test/e2e_node/shortened_grace_period_test.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2enode
+
+import (
+	"context"
+	"github.com/onsi/ginkgo/v2"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	admissionapi "k8s.io/pod-security-admission/api"
+	"time"
+)
+
+var _ = SIGDescribe("Shortened Grace Period", func() {
+	f := framework.NewDefaultFramework("shortened-grace-period")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelBaseline
+	ginkgo.Context("When repeatedly deleting pods", func() {
+		var podClient *e2epod.PodClient
+		ginkgo.BeforeEach(func() {
+			podClient = e2epod.NewPodClient(f)
+		})
+		ginkgo.It("shorter grace period of a second command overrides the longer grace period of a first command", func() {
```

Comment:
```suggestion
		ginkgo.It("shorter grace period of a second command overrides the longer grace period of a first command [NodeConformance]", func() {
```

I think this deserves a conformance label.

Reply:
I think we need to add this label to trigger this e2e test in presubmit tests so that we can ensure that this PR fixes the problem.

Reply:
What is the function of this label

Reply:
https://github.com/kubernetes/test-infra/blob/5f0ff2c9df5deb195a77d219f1aedeb9775299ec/config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml#L76

With this label, this test runs in pull-kubernetes-node-e2e-containerd.

Reply:
I don't quite understand the reason for this one, my understanding of e2e testing is rather weak. I will first follow your method to change, I hope you can help me to test together

Reply:
After adding this, the prompt :timed out while waiting for pod

Reply:
Now we know this test does not work as expected.
We have to fix this test, not excluded it from the tests.

The test fails in L45 as the pod is not ready for 5 minutes.

Reply:
Sorry, I probably couldn't have contributed better to this test, I know too little about e2e

Reply:
I tested the spec in my local environment and figured out that sleep infinity does not work for registry.k8s.io/e2e-test-images/busybox:1.29-4.

You can use sleep 3600 instead.

```sh
$ kubectl logs test
sleep: invalid number 'infinity'
```

Reply:
OK, I'll give it a try